### PR TITLE
[Docs] Kubernetes - Adds AWS authentication example 

### PR DESCRIPTION
**Pull request by askmeegs**
## Hey, I just made a Pull Request!

Adds AWS authentication information, and a configuration snippet, for the Backstage Kubernetes plugin. https://backstage.io/docs/features/kubernetes/authentication#server-side-providers

Closes https://github.com/backstage/backstage/issues/16677 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jamieklassen — APPROVED** [review]
LGTM, nice! I've sometimes heard the access-key/secret-access-key/session-token together described as ["AWS security credentials"](https://docs.aws.amazon.com/general/latest/gr/aws-security-credentials.html). Not saying I have any problem with "IAM keys", though; in my experience these are always things I have gotten from the ["access keys"](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-about) section of the  IAM panel.

**Comment by askmeegs** [comment]
Thanks for the guidance Jamie, updated!

File: docs/features/kubernetes/authentication.md [hunk 1]
```diff
@@ -55,6 +55,37 @@ To get the API server address for your Azure cluster, go to the Azure console pa
...
+          authProvider: 'aws'
+          skipTLSVerify: true
+          skipMetricsLookup: true
+auth:
+  environment: 'production'
+  providers:
+    aws:
+      development:
+        accessKeyId: ${AWS_ACCESS_KEY_ID}
+        secretAccessKey: ${AWS_SECRET_ACCESS_KEY}
```
Comment (Rugvip) [inline docs/features/kubernetes/authentication.md:84]: Is this supposed to be under `aws.accounts` or `aws.mainAccount` instead? per https://github.com/backstage/backstage/blob/master/packages/integration-aws-node/README.md#backstage-app-configuration

As far as I see this config path doesn't exist

**Review by jamieklassen — APPROVED** [review]

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Just a Q about the config structure

**Pull request closed without merging**